From 7d8cce15d329617b717e4b48fc15054f6c8a2bc2 Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@chromium.org>
Date: Tue, 28 Jun 2022 02:29:02 +0000
Subject: [PATCH 44/59] testing: fix check_static_initializers.py for python3
 for M102

(cherry picked from commit dce7a1bfd410fbeb4e17fc6562c150f5dd81310e)

Bug: 1333950
Change-Id: I876378a3ccdbb795c64172555766c027d8259f99
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3689103
Reviewed-by: Brian Sheedy <bsheedy@chromium.org>
Reviewed-by: Ben Pastene <bpastene@chromium.org>
Auto-Submit: Takuto Ikuta <tikuta@chromium.org>
Commit-Queue: Takuto Ikuta <tikuta@chromium.org>
Cr-Original-Commit-Position: refs/heads/main@{#1011772}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3713705
Cr-Commit-Position: refs/branch-heads/5005@{#1203}
Cr-Branched-From: 5b4d9450fee01f821b6400e947b3839727643a71-refs/heads/main@{#992738}
---
 testing/scripts/check_static_initializers.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/testing/scripts/check_static_initializers.py b/testing/scripts/check_static_initializers.py
index 3afd112883b5..d5351e749254 100755
--- a/testing/scripts/check_static_initializers.py
+++ b/testing/scripts/check_static_initializers.py
@@ -64,7 +64,7 @@ COVERAGE_BUILD_FALLBACK_EXPECTED_MAC_SI_COUNT = 4
 
 
 def run_process(command):
-  p = subprocess.Popen(command, stdout=subprocess.PIPE)
+  p = subprocess.Popen(command, stdout=subprocess.PIPE, universal_newlines=True)
   stdout = p.communicate()[0]
   if p.returncode != 0:
     raise Exception(
@@ -205,7 +205,7 @@ def main_run(args):
   if sys.platform.startswith('darwin'):
     rc = main_mac(src_dir,
       allow_coverage_initializer = '--allow-coverage-initializer' in args.args)
-  elif sys.platform == 'linux2':
+  elif sys.platform.startswith('linux'):
     is_chromeos = 'buildername' in args.properties and \
         'chromeos' in args.properties['buildername']
     rc = main_linux(src_dir, is_chromeos)
@@ -222,7 +222,7 @@ def main_run(args):
 def main_compile_targets(args):
   if sys.platform.startswith('darwin'):
     compile_targets = ['chrome']
-  elif sys.platform == 'linux2':
+  elif sys.platform.startswith('linux'):
     compile_targets = ['chrome', 'nacl_helper', 'nacl_helper_bootstrap']
   else:
     compile_targets = []
-- 
2.37.0

